docs/architecture/domain.md

```diff
@@ -0,0 +1,116 @@
+## Definition Model
+
+![Definition](assets/definition-model.png)
+
+### Pipeline
+
+A pipeline represents a collection of [jobs] that share the same source code. These jobs are executed in the order defined by the [workflow]. Only one job is required to be defined in every pipeline:
+
+ - `main` - Job that runs on each change to the source code.  _Note: this is also the same job that runs on proposed changes via GitHub Pull Requests_
+
+### Workflow
+
+Workflow is the defined order that [jobs] will execute in. Jobs can be executed in parallel, series, and a combination of the two to allow for all possibilities. All jobs executed in a given workflow run share:
+
+ - Source code checked out to the same git commit from the `main` [build] that triggered this
+ - Access to the [metadata] from the `main` [build] that triggered this
```

Comment:
Updated.  But in terms of the human trigger, it would be just like a normal commit if starting the `main` job.  But for other jobs, the users would have to pick a previous `main` build to base it on.